CIP-0062/README.md

```diff
@@ -0,0 +1,417 @@
+
+## Error Types
+
+### Extended APIError
+
+```js
+APIErrorCode {
+  UnsupportedVotingPurpose: -100,
+  InvalidArgumentError: -101
+  UnknownChoiceError: -102
+  InvalidBlockDateError: -103
+  InvalidVotePlanError: -104
+  InvalidVoteOptionError: -105
+}
+
+APIError {
+  code: APIErrorCode,
+  info: string
+  votingPurpose: Purpose[]
+}
+```
+
+These are the extended API Error Codes used by this specification. See
+[CIP-30 Errors](https://cips.cardano.org/cips/cip30/#apierror) for the standard
+API Error Codes, which are continue to be valid for this API Extension.
+
+* UnsupportedVotingPurpose - The wallet does not support one of the requested
+  voting purposes. The `votingPurpose` element will be present in the `APIError` object and will list all the voting purposes requested, which are unsupported bgy the wallet. Eg:
+
+    If Voting purposes 0,5 and 9 were requested, and only 0 was supported by the
+    wallet. Then the errors will be:
+
+    ```js
+    {
+      code: -100,
+      info: "Unsupported Voting Purpose 5 & 9",
+      votingPurpose: [5,9]
+    }
+    ```
+
+* InvalidArgumentError - Generic error for errors in the formatting of the arguments.
+* UnknownChoiceError - If a `choice` is not within the
+  [`proposal`](#proposal) option set.
+* InvalidBlockDateError - If a [block date](#blockdate) is invalid.
+* InvalidVotePlanError - If the `votePlanId` is not a valid vote plan.
+* InvalidVoteOptionError - If the `index` is not a valid vote option.
+
+#### APIError elements
+
+* code - The `APIErrorCode` which is being reported.
+* info - A Human readable description of the error.
+* votingPurpose - (*OPTIONAL*) - Only present if the error relates to a voting purpose, and will list all the voting purposes which are involved in the error.  See the individual error code descriptions for more information.
+* rejectedVotes - (*OPTIONAL*) - In a voting transaction, there may be multiple votes being signed simultaneously.
+
+### Extended TxSignError
+
+```js
+TxSignErrorCode {
+  ProofGeneration: 1,
+  UserDeclined: 2,
+  VoteRejected: 3,
+}
+```
+
+```js
+type TxSignError = {
+  code: TxSignErrorCode,
+  info: String,
+  rejectedVotes: number[]
+}
+```
+
+All TxSignErrors defined in [CIP-30](https://cips.cardano.org/cips/cip30/#txsignerror) are unchanged.
+
+* UserDeclined - Raised when the user declined to sign the entire transaction, in the case of a vote submission, this would be returned if the user declined to sign ALL of the votes.
+* VoteRejected - On a vote transaction, where there may be multiple votes.  If the user accepted some votes, but rejected others, then this error is raised, AND `rejectedVotes` is present in the Error instance.
+
+## Governance Extension to CIP-30
+
+### cardano.{walletName}.governance.apiVersion: String
+
+The version number of the Governance Extension API that the wallet supports.
+
+### cardano.{walletName}.governance.enable(purpose: VotingPurpose[]): Promise\<API>
```

Comment:
This is to remove ambiguity of in-what context is the dapp requesting connection to the wallet. The prompt to the user should be clarify for application, use case is requesting connection. Otherwise after the whitelist, potentially the dApp could use it for other means. This was a separate method call, but through discussions and suggestion of @stevenj , it was included in the `enable` method.